<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Badawi" failures="0" tests="10" name="DolingoMobile Testing Project" time="27.652" errors="0" timestamp="2025-10-07T20:24:19 EEST">
  <testcase classname="base.TestBase" name="@BeforeClass setUp" time="20.59">
    <failure type="org.openqa.selenium.SessionNotCreatedException" message="Unable to create a new remote session. Please check the server log for more details. Original error: Could not find a connected Android device in 20000ms
Build info: version: &amp;apos;3.141.59&amp;apos;, revision: &amp;apos;e82be7d358&amp;apos;, time: &amp;apos;2018-11-14T08:17:03&amp;apos;
System info: host: &amp;apos;BADAWI&amp;apos;, ip: &amp;apos;192.168.1.137&amp;apos;, os.name: &amp;apos;Windows 11&amp;apos;, os.arch: &amp;apos;amd64&amp;apos;, os.version: &amp;apos;10.0&amp;apos;, java.version: &amp;apos;24.0.2&amp;apos;
Driver info: driver.version: AndroidDriver
remote stacktrace: UnknownError: Could not find a connected Android device in 20000ms
    at getResponseForW3CError (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\errors.ts:1103:34)
    at asyncHandler (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.ts:507:57)
The above error is caused by
Error: Could not find a connected Android device in 20000ms
    at ADB.getDevicesWithRetry (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-adb\lib\tools\system-calls.js:296:13)
    at AndroidUiautomator2Driver.getDeviceInfoFromCaps (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\commands\device\common.js:41:21)
    at AndroidUiautomator2Driver.createSession (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\lib\driver.ts:361:30)
    at AppiumDriver.createSession (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\lib\appium.js:360:35)
Build info: version: &amp;apos;3.141.59&amp;apos;, revision: &amp;apos;e82be7d358&amp;apos;, time: &amp;apos;2018-11-14T08:17:03&amp;apos;
System info: host: &amp;apos;BADAWI&amp;apos;, ip: &amp;apos;192.168.1.137&amp;apos;, os.name: &amp;apos;Windows 11&amp;apos;, os.arch: &amp;apos;amd64&amp;apos;, os.version: &amp;apos;10.0&amp;apos;, java.version: &amp;apos;24.0.2&amp;apos;
Driver info: driver.version: AndroidDriver">
      <![CDATA[org.openqa.selenium.SessionNotCreatedException: Unable to create a new remote session. Please check the server log for more details. Original error: Could not find a connected Android device in 20000ms
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'BADAWI', ip: '192.168.1.137', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '24.0.2'
Driver info: driver.version: AndroidDriver
remote stacktrace: UnknownError: Could not find a connected Android device in 20000ms
    at getResponseForW3CError (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\errors.ts:1103:34)
    at asyncHandler (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.ts:507:57)
The above error is caused by
Error: Could not find a connected Android device in 20000ms
    at ADB.getDevicesWithRetry (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-adb\lib\tools\system-calls.js:296:13)
    at AndroidUiautomator2Driver.getDeviceInfoFromCaps (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\commands\device\common.js:41:21)
    at AndroidUiautomator2Driver.createSession (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\lib\driver.ts:361:30)
    at AppiumDriver.createSession (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\lib\appium.js:360:35)
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'BADAWI', ip: '192.168.1.137', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '24.0.2'
Driver info: driver.version: AndroidDriver
at io.appium.java_client.remote.AppiumCommandExecutor$1.createSession(AppiumCommandExecutor.java:216)
at io.appium.java_client.remote.AppiumCommandExecutor.createSession(AppiumCommandExecutor.java:225)
at io.appium.java_client.remote.AppiumCommandExecutor.execute(AppiumCommandExecutor.java:250)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:552)
at io.appium.java_client.DefaultGenericMobileDriver.execute(DefaultGenericMobileDriver.java:41)
at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:1)
at io.appium.java_client.android.AndroidDriver.execute(AndroidDriver.java:1)
at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:213)
at io.appium.java_client.AppiumDriver.startSession(AppiumDriver.java:363)
at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
at io.appium.java_client.DefaultGenericMobileDriver.<init>(DefaultGenericMobileDriver.java:37)
at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:89)
at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:99)
at io.appium.java_client.android.AndroidDriver.<init>(AndroidDriver.java:97)
at base.TestBase.setUp(TestBase.java:40)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at io.appium.java_client.remote.AppiumCommandExecutor$1.createSession(AppiumCommandExecutor.java:193)
... 38 more
Caused by: org.openqa.selenium.WebDriverException: Could not find a connected Android device in 20000ms
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'BADAWI', ip: '192.168.1.137', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '24.0.2'
Driver info: driver.version: AndroidDriver
remote stacktrace: UnknownError: Could not find a connected Android device in 20000ms
    at getResponseForW3CError (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\errors.ts:1103:34)
    at asyncHandler (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.ts:507:57)
The above error is caused by
Error: Could not find a connected Android device in 20000ms
    at ADB.getDevicesWithRetry (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-adb\lib\tools\system-calls.js:296:13)
    at AndroidUiautomator2Driver.getDeviceInfoFromCaps (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\commands\device\common.js:41:21)
    at AndroidUiautomator2Driver.createSession (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\lib\driver.ts:361:30)
    at AppiumDriver.createSession (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\lib\appium.js:360:35)
at org.openqa.selenium.remote.W3CHandshakeResponse.lambda$errorHandler$0(W3CHandshakeResponse.java:62)
at org.openqa.selenium.remote.HandshakeResponse.lambda$getResponseFunction$0(HandshakeResponse.java:30)
at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:126)
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1034)
at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:687)
at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:128)
... 40 more
... Removed 30 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeClass setUp -->
  <testcase classname="base.TestBase" name="@AfterClass tearDown" time="0.0">
    <skipped/>
  </testcase> <!-- @AfterClass tearDown -->
  <testcase classname="base.TestBase" name="@BeforeClass setUp" time="6.964">
    <failure type="org.openqa.selenium.SessionNotCreatedException" message="Unable to create a new remote session. Please check the server log for more details. Original error: Error executing adbExec. Original error: &amp;apos;Command &amp;apos;&amp;apos;C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe&amp;apos; -P 5037 -s emulator-5554 shell &amp;apos;settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy&amp;apos;&amp;apos; exited with code 20&amp;apos;; Command output: cmd: Can&amp;apos;t find service: settings
cmd: Can&amp;apos;t find service: settings
cmd: Can&amp;apos;t find service: settings

Build info: version: &amp;apos;3.141.59&amp;apos;, revision: &amp;apos;e82be7d358&amp;apos;, time: &amp;apos;2018-11-14T08:17:03&amp;apos;
System info: host: &amp;apos;BADAWI&amp;apos;, ip: &amp;apos;192.168.1.137&amp;apos;, os.name: &amp;apos;Windows 11&amp;apos;, os.arch: &amp;apos;amd64&amp;apos;, os.version: &amp;apos;10.0&amp;apos;, java.version: &amp;apos;24.0.2&amp;apos;
Driver info: driver.version: AndroidDriver
remote stacktrace: UnknownError: Error executing adbExec. Original error: &amp;apos;Command &amp;apos;&amp;apos;C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe&amp;apos; -P 5037 -s emulator-5554 shell &amp;apos;settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy&amp;apos;&amp;apos; exited with code 20&amp;apos;; Command output: cmd: Can&amp;apos;t find service: settings
cmd: Can&amp;apos;t find service: settings
cmd: Can&amp;apos;t find service: settings

    at getResponseForW3CError (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\errors.ts:1103:34)
    at asyncHandler (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.ts:507:57)
The above error is caused by
Error: Error executing adbExec. Original error: &amp;apos;Command &amp;apos;&amp;apos;C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe&amp;apos; -P 5037 -s emulator-5554 shell &amp;apos;settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy&amp;apos;&amp;apos; exited with code 20&amp;apos;; Command output: cmd: Can&amp;apos;t find service: settings
cmd: Can&amp;apos;t find service: settings
cmd: Can&amp;apos;t find service: settings

    at ChildProcess.&amp;lt;anonymous&amp;gt; (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\teen_process\lib\exec.js:110:19)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1101:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
    at Process.callbackTrampoline (node:internal/async_hooks:130:17)
Build info: version: &amp;apos;3.141.59&amp;apos;, revision: &amp;apos;e82be7d358&amp;apos;, time: &amp;apos;2018-11-14T08:17:03&amp;apos;
System info: host: &amp;apos;BADAWI&amp;apos;, ip: &amp;apos;192.168.1.137&amp;apos;, os.name: &amp;apos;Windows 11&amp;apos;, os.arch: &amp;apos;amd64&amp;apos;, os.version: &amp;apos;10.0&amp;apos;, java.version: &amp;apos;24.0.2&amp;apos;
Driver info: driver.version: AndroidDriver">
      <![CDATA[org.openqa.selenium.SessionNotCreatedException: Unable to create a new remote session. Please check the server log for more details. Original error: Error executing adbExec. Original error: 'Command ''C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe' -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 20'; Command output: cmd: Can't find service: settings
cmd: Can't find service: settings
cmd: Can't find service: settings

Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'BADAWI', ip: '192.168.1.137', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '24.0.2'
Driver info: driver.version: AndroidDriver
remote stacktrace: UnknownError: Error executing adbExec. Original error: 'Command ''C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe' -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 20'; Command output: cmd: Can't find service: settings
cmd: Can't find service: settings
cmd: Can't find service: settings

    at getResponseForW3CError (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\errors.ts:1103:34)
    at asyncHandler (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.ts:507:57)
The above error is caused by
Error: Error executing adbExec. Original error: 'Command ''C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe' -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 20'; Command output: cmd: Can't find service: settings
cmd: Can't find service: settings
cmd: Can't find service: settings

    at ChildProcess.<anonymous> (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\teen_process\lib\exec.js:110:19)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1101:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
    at Process.callbackTrampoline (node:internal/async_hooks:130:17)
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'BADAWI', ip: '192.168.1.137', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '24.0.2'
Driver info: driver.version: AndroidDriver
at io.appium.java_client.remote.AppiumCommandExecutor$1.createSession(AppiumCommandExecutor.java:216)
at io.appium.java_client.remote.AppiumCommandExecutor.createSession(AppiumCommandExecutor.java:225)
at io.appium.java_client.remote.AppiumCommandExecutor.execute(AppiumCommandExecutor.java:250)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:552)
at io.appium.java_client.DefaultGenericMobileDriver.execute(DefaultGenericMobileDriver.java:41)
at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:1)
at io.appium.java_client.android.AndroidDriver.execute(AndroidDriver.java:1)
at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:213)
at io.appium.java_client.AppiumDriver.startSession(AppiumDriver.java:363)
at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
at io.appium.java_client.DefaultGenericMobileDriver.<init>(DefaultGenericMobileDriver.java:37)
at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:89)
at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:99)
at io.appium.java_client.android.AndroidDriver.<init>(AndroidDriver.java:97)
at base.TestBase.setUp(TestBase.java:40)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at io.appium.java_client.remote.AppiumCommandExecutor$1.createSession(AppiumCommandExecutor.java:193)
... 38 more
Caused by: org.openqa.selenium.WebDriverException: Error executing adbExec. Original error: 'Command ''C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe' -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 20'; Command output: cmd: Can't find service: settings
cmd: Can't find service: settings
cmd: Can't find service: settings

Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'BADAWI', ip: '192.168.1.137', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '24.0.2'
Driver info: driver.version: AndroidDriver
remote stacktrace: UnknownError: Error executing adbExec. Original error: 'Command ''C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe' -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 20'; Command output: cmd: Can't find service: settings
cmd: Can't find service: settings
cmd: Can't find service: settings

    at getResponseForW3CError (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\errors.ts:1103:34)
    at asyncHandler (C:\Users\moham\AppData\Roaming\npm\node_modules\appium\node_modules\@appium\base-driver\lib\protocol\protocol.ts:507:57)
The above error is caused by
Error: Error executing adbExec. Original error: 'Command ''C:\Users\moham\AppData\Local\Android\Sdk\platform-tools\adb.exe' -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 20'; Command output: cmd: Can't find service: settings
cmd: Can't find service: settings
cmd: Can't find service: settings

    at ChildProcess.<anonymous> (C:\Users\moham\.appium\node_modules\appium-uiautomator2-driver\node_modules\teen_process\lib\exec.js:110:19)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1101:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
    at Process.callbackTrampoline (node:internal/async_hooks:130:17)
at org.openqa.selenium.remote.W3CHandshakeResponse.lambda$errorHandler$0(W3CHandshakeResponse.java:62)
at org.openqa.selenium.remote.HandshakeResponse.lambda$getResponseFunction$0(HandshakeResponse.java:30)
at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:126)
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:1034)
at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:687)
at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:128)
... 40 more
... Removed 30 stack frames]]>
    </failure>
  </testcase> <!-- @BeforeClass setUp -->
  <testcase classname="base.TestBase" name="@AfterClass tearDown" time="0.0">
    <skipped/>
  </testcase> <!-- @AfterClass tearDown -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="AddDemoTasks" time="0.0">
    <skipped/>
  </testcase> <!-- AddDemoTasks -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="DeleteTask" time="0.0">
    <skipped/>
  </testcase> <!-- DeleteTask -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="CancelDeleteTask" time="0.0">
    <skipped/>
  </testcase> <!-- CancelDeleteTask -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="DeleteFromTask" time="0.0">
    <skipped/>
  </testcase> <!-- DeleteFromTask -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="DeleteMultipleTasks" time="0.001">
    <skipped/>
  </testcase> <!-- DeleteMultipleTasks -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="DeleteAllTask" time="0.0">
    <skipped/>
  </testcase> <!-- DeleteAllTask -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="ClearCompleteTask" time="0.0">
    <skipped/>
  </testcase> <!-- ClearCompleteTask -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="DeleteMixedTask" time="0.0">
    <skipped/>
  </testcase> <!-- DeleteMixedTask -->
  <testcase classname="Badawi_Test.TS3_DeleteTask" name="DeleteFromSearch" time="0.0">
    <skipped/>
  </testcase> <!-- DeleteFromSearch -->
  <testcase classname="Badawi_Test.TS4_CompleteTask" name="testAddSimpleTask" time="0.0">
    <skipped/>
  </testcase> <!-- testAddSimpleTask -->
</testsuite> <!-- DolingoMobile Testing Project -->
